MCKIN-7788 Adds Google Analytics click handlers to PDF download links

pomegranited opened the pull request:
Sends an event to Google Analytics (if configured) when a learner clicks on either the 
"PDF Report" or "Key Takeaways" download links displayed by this xblock.

**Screenshots**
EOC Journal XBlock GUI (unchanged):
![eoc journal](https://user-images.githubusercontent.com/7556571/42016239-ed12c1e0-7ae9-11e8-82a4-bbf2060fea51.png)

Google analytics screen showing events received for these events:
![ga events](https://user-images.githubusercontent.com/7556571/42068181-a594f90a-7b89-11e8-9425-1a2ebcff0677.png)

**Testing instructions**
1. Create a course with a subsection containing a Problem Builder Long answer, and an EOC Journal unit, e.g. [MCKIN-7788.ooVv6d.tar.gz](https://github.com/open-craft/xblock-eoc-journal/files/2147388/MCKIN-7788.ooVv6d.tar.gz)
  Ensure that a Key Takeaways PDF is provided, and/or the EOC unit is linked to the Problem Builder long answer problem.
* View the PB unit in Studio, LMS, or Apros, and complete the long answer problem.
  This allows the EOC XBlock to show the "PDF Report" download link.
* View the EOC unit in Apros without enabling Google Analytics.
   Note that no errors should be seen in the JS console, and the EOC XBlock should continue to function as usual.
* Enable Google Analytics by updating the apros `settings.GA_TRACKING_ID` to a tracking ID (`UA-xxxxxxxxx-x`).
* Reload the EOC unit, and note that an event is sent to Google Analytics when the PDF download links are clicked.
   * *Note*: Don't use a private browsing window to test the GA hit -- browsers like [Firefox provide tracking protection](https://developer.mozilla.org/en-US/Firefox/Privacy/Tracking_Protection?utm_source=mozilla&utm_medium=firefox-console-errors&utm_campaign=default) against domains which track users across sites, and so the GA script won't load.

**Reviewers**
- [ ] @symbolist

symbolist commented:
@pomegranited Any reason you haven't used the `initialize_js` hook? (for example: https://github.com/edx/edx-ora2/blob/df4f49a4cb7a6feadbf9444a95ce11d81cd216b9/openassessment/xblock/openassessmentblock.py#L534)

And can you test the following scenario:
1. Add a vertical with a html block before the vertical which has the EOC block. So that subsection will have two verticals: [Vertical with html block, Vertical with an EOC block].
2. Visit the subsection and then switch to the vertical with the EOC block.
3. Do the events get triggered?

pomegranited commented:
@symbolist  Ah oops -- I totally forgot about the `initialize_js` hook. 😊  Thank you for catching that!  
ef20280 addresses that issue, and I've updated the attached course export to use a course which properly tests this behaviour.

Unfortunately, I haven't been able to make this code work with the [GA scripts as they are currently loaded by the LMS template](https://github.com/edx/edx-platform/blob/master/lms/templates/main.html#L148-L158).  To get this to work in the LMS, I had to change them to load GA like Apros is doing:
```diff
diff --git a/lms/templates/main.html b/lms/templates/main.html
index 31a45f8012..ed8e5eadf7 100644
--- a/lms/templates/main.html
+++ b/lms/templates/main.html
@@ -113,14 +113,12 @@ from pipeline_mako import render_require_js_path_overrides
 <% ga_acct = static.get_value("GOOGLE_ANALYTICS_ACCOUNT", settings.GOOGLE_ANALYTICS_ACCOUNT) %>
 % if ga_acct:
     <script type="text/javascript">
-    var _gaq = _gaq || [];
-    _gaq.push(['_setAccount', '${ga_acct | n, js_escaped_string}']);
-    _gaq.push(['_trackPageview']);
-    (function() {
-      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
-      ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
-      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
-    })();
+        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
+          (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
+          m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+          })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
+          ga('create', '${ga_acct | n, js_escaped_string}', 'auto');
+          ga('send', 'pageview');
     </script>
 % endif
```

But since this change only needs to work in Apros, I think this is ok.  If you finish the code review @symbolist using the `master` LMS as altered above, maybe @xitij2000 could verify it in Apros?

Have also changed the event category to "Course Journal" match the updated XBlock displayed title as altered by https://github.com/open-craft/xblock-eoc-journal/pull/14.

pomegranited commented:
@xitij2000  verified this on his Apros devstack, so will merge now.
